File: Solutions/0242- Valid Anagram/Solution.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())
            return false;

        HashMap<Character, Integer> map = new HashMap<>();

        for (var ch : s.toCharArray()) {
            if (map.containsKey(ch))
                map.put(ch, map.get(ch) + 1);
            else
                map.put(ch, 1);

        }

        for (var ch : t.toCharArray()) {
            if (!map.containsKey(ch))
                return false;

            if(map.get(ch) > 1)
                map.put(ch, map.get(ch) - 1);
            else
                map.remove(ch);

        }

        return true;

    }
}

File: Solutions/0242- Valid Anagram/Solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        map = {}

        for ch in s:
            if map.get(ch):
                map[ch] += 1
            else:
                map[ch] = 1

        for ch in t:
            if not map.get(ch):
                return False

            if map[ch] > 1:
                map[ch] -= 1
            else:
                del map[ch]

        return True